cmd/tailscaled/tailscaled.go

```diff
@@ -257,16 +257,22 @@ func main() {
 	// Only apply a default statepath when neither have been provided, so that a
 	// user may specify only --statedir if they wish.
 	if args.statepath == "" && args.statedir == "" {
-		if runtime.GOOS == "plan9" {
+		setMakeStateDir := func(d string) {
+			args.statedir = d
+			if err := os.MkdirAll(d, 0700); err != nil {
+				log.Fatalf("failed to create state directory: %v", err)
+			}
+		}
+		switch {
+		case runtime.GOOS == "plan9":
 			home, err := os.UserHomeDir()
 			if err != nil {
 				log.Fatalf("failed to get home directory: %v", err)
 			}
-			args.statedir = filepath.Join(home, "tailscale-state")
-			if err := os.MkdirAll(args.statedir, 0700); err != nil {
-				log.Fatalf("failed to create state directory: %v", err)
-			}
-		} else {
+			setMakeStateDir(filepath.Join(home, "tailscale-state"))
+		case runtime.GOOS == "linux" && distro.Get() == distro.JetKVM:
+			setMakeStateDir("/userdata/tailscale/var")
```

Comment:
can this move to `paths.DefaultTailscaledStateFile`?
maybe plan9 stuff too

Reply:
the directory can still get created here if you'd like (although I'm pretty sure `store.New` creates the parent directory as needed already). but all the path guessing logic doesn't need to be spread across several packages.